test/unit/node/routes/vscode.test.ts

```diff
@@ -136,6 +99,42 @@ describe("vscode", () => {
     await resp.text()
   })
 
+  it("should add the workspace as a query param maintaining the slashes", async () => {
+    const workspace = path.join(await tmpdir(testName), "test.code-workspace")
+    await fs.writeFile(workspace, "")
+    codeServer = await integration.setup(["--auth=none", workspace], "")
+
+    let resp = await codeServer.fetch("/", undefined)
+
+    expect(resp.status).toBe(200)
+    const url = new URL(resp.url)
+    expect(url.search).toBe(`?workspace=${workspace}`)
+    await resp.text()
+  })
```

Comment:
This is an interesting case.  While it is the  same test as the one above the description is different and it does bring attention to a different thing even if the code is identical.

Should we keep it or try renaming the other?  For example we could rename the other to something like `should redirect to the passed in workspace using human-readable query` which covers both the redirect case and the human-readable case.

Reply:
I'll rename it to the other - good suggestion!